### PR TITLE
implemented mock login

### PR DESCRIPTION
**Pull request by aSvetenco**

### CODE REVIEW

File: MeetUp/app/src/main/java/com/endava/meetup/data/UserParser.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            stream.read(buffer);
+            stream.close();
+            json = new String(buffer, "UTF-8");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return serializeFromJson(json);
+    }
+
+    private List<User> serializeFromJson(String json) {
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/data/UserParser.java]: you are making deserialization, not serialize

Comment (UncuSergiu) [reply MeetUp/app/src/main/java/com/endava/meetup/data/UserParser.java]: i think you could use gson or something like that

File: MeetUp/app/src/main/java/com/endava/meetup/data/model/User.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private static final String EMAIL_PREFIX = "email";
+    private static final String PASSWORD_PREFIX = "password";
+    private static final String API_KEY_PREFIX = "api_key";
+    public static final String[] prefixes = {NAME_PREFIX, EMAIL_PREFIX, PASSWORD_PREFIX, API_KEY_PREFIX};
+    private String userName;
+    private String email;
+    private String password;
+    private String apiKey;
+
+    public User() {
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/data/model/User.java]: do you need the empty constructor?

File: MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/signin/SignInFragment.java [hunk 3]
```diff
@@ -43,23 +57,29 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
 
         mEmail = view.findViewById(R.id.signin_input_email);
         mPassword = view.findViewById(R.id.signin_input_password);
-
         emailInput = view.findViewById(R.id.edittext_email_signin);
         passwordInput = view.findViewById(R.id.edittext_password_signin);
+        loaderContainer = view.findViewById(R.id.loader_container);
 
         button = view.findViewById(R.id.btn_login);
         button.setOnClickListener(new View.OnClickListener() {
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/signin/SignInFragment.java]: you can get the click listener from a presenter method, by passing the needed params

File: MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/signin/SignInPresenter.java [hunk 4]
```diff
@@ -57,10 +65,41 @@ void onPasswordTextChanged(CharSequence data) {
...
         return isEmptyValidator.isValid(data);
     }
+
     private void manageButtonStatePresenter() {
         getView().manageButtonState(isEmailValid && isPasswordValid);
     }
+
+    void onSignInButtonClick(AssetsAsyncTask assetsAsyncTask, final String email, final String password) {
+        getView().showLoading();
+        assetsAsyncTask.setOnPostExecuteListener(new AsyncTaskCallback<List<?>>() {
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/signin/SignInPresenter.java]: you don't have to check the credentials in AsyncTask. This is needed just for heavy load operations, like extracting from assets. A regular check is enough

File: MeetUp/app/src/main/res/layout/layout_loader_container.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (CebotariM) [inline MeetUp/app/src/main/res/layout/layout_loader_container.xml]: why do you need a LinearLayout inside a FrameLayout?

File: MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/signin/SignInModel.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    private final AssetsHelper assetsHelper;
+    private final InputStreamParser<List<User>> userParser;
+
+    public SignInModel(String path, AssetsHelper assetsHelper, InputStreamParser userParser) {
+        this.path = path;
+        this.assetsHelper = assetsHelper;
+        this.userParser = userParser;
+    }
+
+    public List<User> getUserListFromAssets() {
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/signin/SignInModel.java]: it is usually bad practice to put logic inside methods. Please move this method to some extractor class

File: MeetUp/app/src/main/java/com/endava/meetup/data/AssetsHelper.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package com.endava.meetup.data;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+public class AssetsHelper {
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/data/AssetsHelper.java]: by name, this class should handle all the assets work, you just passing to it a callback, and it handling asyncTask and the rest

File: MeetUp/app/src/main/res/layout/layout_loader_container.xml (hunk 5 above)
Comment (aSvetenco) [reply MeetUp/app/src/main/res/layout/layout_loader_container.xml]: fixed

File: MeetUp/app/src/main/java/com/endava/meetup/data/model/User.java (hunk 2 above)
Comment (aSvetenco) [reply MeetUp/app/src/main/java/com/endava/meetup/data/model/User.java]: fixed

**Pull request merged**